### PR TITLE
test: fatal exception: add test case

### PR DESCRIPTION
**Pull request by yingmingx**
Add regex in test.yaml to verify the kernel will dump
information when in some exception conditions.

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

File: tests/kernel/fatal/testcase.yaml [hunk 1]
```diff
@@ -1,3 +1,11 @@
+common:
+    tags: tracing debug
+    harness: console
+    harness_config:
+      type: multi_line
```
Comment (nashif) [inline tests/kernel/fatal/testcase.yaml:5]: not sure why you are adding this, this test is a ztest and the output is already evaluated... You should not need this.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/kernel/fatal/testcase.yaml (hunk 1 above)
Comment (yingmingx) [reply tests/kernel/fatal/testcase.yaml:5]: I want to verify weather the zephyr OS would dump faulting thread ID number and the error type.  In source code,  LOG_ERR and TC_PRINT were used to print this information.  So I referenced samples/hello_world/sample.yaml and modify the yaml in tests/kernel/fatal/testcase.yaml.
Do you mean the file in tests is different from the file in samples or I should only left 
regex:
        - "Current thread: (.*)!"
        - "Caught system error -- reason (.*)!"
and erase the common tags?

Comment (nashif) [reply tests/kernel/fatal/testcase.yaml:5]: > so there is no need to use regex to verify the print information or I should only left

here we verify pass/fail based on the output from the test already, ie.. we parse pass/fail information, so there is no need for additional regex to capture that.

Comment (yingmingx) [reply tests/kernel/fatal/testcase.yaml:5]: thanks, I will close this PR.

Comment (enjiamai) [reply tests/kernel/fatal/testcase.yaml:5]: @nashif In this case, YingMing want to verify if z_fatal_error will generate output when fatal error happened. (which locates in kernel/fatal.c +105  and +120, output by LOG_ERR macro). So our method is to try to capture the LOG_ERR output from here to judge the result of error reason and thread ID.  It seems like we can't simply take advantage of zassert here.
Our another idea is to add sys_trace call to verify if LOG_ERR is invoked already, do you think this work?

Comment (LeiW000) [reply tests/kernel/fatal/testcase.yaml:5]: What Yingming wants to verify is not only if the test cases pass/fail, and also if the error messages are shown on console, when fatal error occurs.

**Review by LeiW000 — APPROVED** [review]
It looks good to me.

File: tests/kernel/fatal/testcase.yaml (hunk 1 above)
Comment (nashif) [reply tests/kernel/fatal/testcase.yaml:5]: guys, tests that are using ztest should not try and verify output, this is done already automatically based on test output, we can mix things like this.

**Review by nashif — CHANGES_REQUESTED** [review]

**Comment by yingmingx** [comment]
This PR will be closed because there are some compilation problems.

**Pull request closed without merging**